app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -50,21 +68,25 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
+        // TODO fix this quick fix to on success being called twice
+        // onSuccess for login is being called twice even though login button onClick is called once
+        onSuccessCalled = false;
 
         context = this;
 
         existingPages = new HashMap<>();
         ParseQuery<Page> query = ParseQuery.getQuery(Page.class);
-        query.whereExists("objectId");
+        query.whereExists(OBJECT_ID);
         query.findInBackground(new FindCallback<Page>() {
             @Override
             public void done(List<Page> objects, ParseException e) {
-                if (e == null) {
+                if (objects != null && !objects.isEmpty()) {
                     for (Page page: objects) {
                         existingPages.put(page.getPageId(), page.getObjectId());
```

Comment:
check if page is null? and make sure you are not adding null while putting in the map.

Reply:
Didn't think of that, thanks!